crypto/sha/keccak1600.c

```diff
@@ -11,6 +11,10 @@
 #include <string.h>
 #include <assert.h>
 
+size_t SHA3_absorb(uint64_t A[5][5], const unsigned char *inp, size_t len,
+                   size_t r);
+void SHA3_squeeze(uint64_t A[5][5], unsigned char *out, size_t len, size_t r);
+
```

Comment:
It would be better if you placed these declarations in, say, `crypto/include/internal/sha_int.h` and included "internal/sha_int.h" both here and in m_sha3.c